lib/cluster/index.ts

```diff
@@ -477,11 +479,9 @@ class Cluster extends Commander {
     let targetSlot = node ? node.slot : command.getSlot();
     const ttl = {};
     const _this = this;
-    // @ts-expect-error
-    if (!node && !command.__is_reject_overwritten) {
-      // TODO WeakMap
-      // @ts-expect-error
-      command.__is_reject_overwritten = true;
+    if (!node && !REJECT_OVERWRITTEN_COMMANDS.has(command)) {
```

Comment:
Keep in mind that the `command.promise` & `command.reject` could get overwritten in some cases.
But at the moment it's only done in the [pipeline](https://github.com/luin/ioredis/blob/b45a3e82d9f3daf096803936192b4397a6e05b46/lib/Pipeline.ts#L150) and for that `node` should be present here.
It's probably good enough at the moment, but `.has(command.promise)` might be better.

Reply:
Good point. Will think about it!